airbyte-webapp-e2e-tests/cypress/pages/connection/streamsTablePageObject/StreamRowPageObject.ts

```diff
@@ -0,0 +1,61 @@
+const streamTableRow = (namespace: string, streamName: string) =>
+  `div[data-testid='catalog-tree-table-row-${namespace}-${streamName}']`;
+const streamPanel = (namespace: string, streamName: string) =>
+  `div[data-testid='stream-details-panel-${namespace}-${streamName}']`;
+const streamPanelCloseButton = `button[data-testid='stream-panel-close-button']`;
+const streamSyncSwitch = `label[data-testid='sync-switch']`;
+const sourceStreamNameCell = `div[data-testid='source-stream-name-cell']`;
+const destinationStreamNameCell = `div[data-testid='destination-stream-name-cell']`;
+const sourceNamespaceCell = `div[data-testid='source-namespace-cell']`;
+const destinationNamespaceCell = `div[data-testid='destination-namespace-cell']`;
+
+export class StreamRowPageObject {
+  static getStreamUtilityFunctions(namespace: string, streamName: string) {
+    const stream = streamTableRow(namespace, streamName);
+    const panel = streamPanel(namespace, streamName);
+
+    const isStreamSyncEnabled = (expectedValue: boolean) =>
+      cy.get(stream).within(() => {
+        cy.get(streamSyncSwitch)
+          .get("input")
+          .should(`${expectedValue ? "" : "not."}be.checked`);
+      });
+
+    const toggleStreamSync = () =>
+      cy.get(stream).within(() => {
+        cy.get(streamSyncSwitch).click();
+      });
+
+    const isStreamRowHasRemovedStyle = (expectedValue: boolean) =>
+      cy
+        .get(stream)
+        .invoke("attr", "class")
+        .should(`${expectedValue ? "" : "not."}match`, /removed/);
+
+    const checkSourceNamespace = () => cy.get(stream).within(() => cy.get(sourceNamespaceCell).contains(namespace));
+    const checkSourceStreamName = () => cy.get(stream).within(() => cy.get(sourceStreamNameCell).contains(streamName));
+
+    const checkDestinationNamespace = (expectedValue: string) =>
+      cy.get(stream).within(() => cy.get(destinationNamespaceCell).contains(expectedValue));
+    const checkDestinationStreamName = (expectedValue: string) =>
+      cy.get(stream).within(() => cy.get(destinationStreamNameCell).contains(expectedValue));
+
+    const openStreamPanel = () => cy.get(stream).within(() => cy.get(destinationNamespaceCell).click());
+    const closeStreamPanel = () => cy.get(panel).within(() => cy.get(streamPanelCloseButton).click());
+    const isStreamPanelVisible = (expectedValue: boolean) =>
+      cy.get(panel).should(`${expectedValue ? "" : "not."}exist`);
+
+    return {
+      isStreamSyncEnabled,
+      toggleStreamSync,
+      isStreamRowHasRemovedStyle,
+      checkSourceNamespace,
+      checkSourceStreamName,
+      checkDestinationNamespace,
+      checkDestinationStreamName,
+      openStreamPanel,
+      closeStreamPanel,
+      isStreamPanelVisible,
+    };
+  }
+}
```

Comment:
Just calling out the difference in implementations from what I've seen.
You're exporting a class w/ static methods and Edmundo was exporting a new instance of a class.
I'm good with either implementation but think we should converge!

Reply:
@edmundito seems like I need to convert everything to your approach since there is more code in that style

Reply:
I'd be willing to make that a part 2 of this.

Reply:
Yes, the class for this should be:

```ts
export class StreamRowPageObject {
   constructor(namespace: string, streamName: string) {}
   isStreamSyncEnabled(): boolean {}
   // .. the other functions...
}
```

Reply:
updated